src/storage/data_table.cpp

```diff
@@ -300,7 +300,7 @@ void DataTable::NewBlock(RawBlock *expected_val) {
   // Want to stop early if another thread is already getting a new block
   if (expected_val != insertion_head_) return;
   RawBlock *new_block = block_store_->Get();
-  accessor_.InitializeRawBlock(new_block, layout_version_);
+  accessor_.InitializeRawBlockForDataTable(this, new_block, layout_version_);
```

Comment:
This is literally the only reference to `IntializeRawBlock(ForDataTable)` which means one of the two functions will never be used.  Choose what the function name should be and get rid of the other function.